### PR TITLE
FOGL-8306: Show system alerts

### PR DESCRIPTION
**Pull request by MonikaSharma06**

### CODE REVIEW

File: src/app/components/core/system-alert/system-alert.component.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+  applyClass(urgency: string) {
+    if (urgency.toLowerCase() === "critical") {
+      return "has-text-danger";
+    }
+    if (urgency.toLowerCase() === "high") {
+      return "has-text-warning";
+    }
+    if (urgency.toLowerCase() === "normal") {
+      return "has-text-grey-light";
```
Comment (praveen-garg) [inline src/app/components/core/system-alert/system-alert.component.ts:122]: has-text-warning-light

File: src/app/components/layout/navbar/navbar.component.ts [hunk 2]
```diff
@@ -237,7 +239,7 @@ export class NavbarComponent implements OnInit, AfterViewInit, OnDestroy {
           this.pingInfo.isAuth = true;
         } else {
           this.sharedService.connectionInfo.next({ version: '', isServiceUp: false });
-          this.pingInfo = { isAlive: false, isAuth: false, isSafeMode: false, hostName: '', version: '' };
+          this.pingInfo = { isAlive: false, isAuth: false, isSafeMode: false, hostName: '', version: '', alertCount: this.pingData['alerts'] };
```
Comment (praveen-garg) [inline src/app/components/layout/navbar/navbar.component.ts:242]: alertsCount

File: src/app/services/system-alert.service.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import { environment } from '../../environments/environment';
+
+@Injectable()
+export class SystemAlertService {
+  constructor(private http: HttpClient) { }
+
+  /**
+   *  GET | /fledge/alert
+   */
```
Comment (praveen-garg) [inline src/app/services/system-alert.service.ts:14]: remove?

File: src/app/components/core/system-alert/system-alert.component.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    if (urgency.toLowerCase() === "low") {
+      return "text-grey-lighter";
```
Comment (praveen-garg) [inline src/app/components/core/system-alert/system-alert.component.ts:125]: has-text-info-light

File: src/app/components/core/system-alert/system-alert.component.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        }
+      }
+    );
+  }
+
+  getButtonText(message: string) {
+    if (message.includes('restarted')) {
+      return "Show Logs";
+    }
+    if (message.includes('updates')) {
```
Comment (praveen-garg) [inline src/app/components/core/system-alert/system-alert.component.ts:109]: upgrade

**Review by praveen-garg — COMMENTED** [review]
With these fixes, We can merge as part-1

And rest of the things like group by, sorting, Hide all etc. can be done in next FOGL-8306.2 PR, just next.

File: src/app/components/core/system-alert/system-alert.component.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+      return "Upgrade";
```
Comment (praveen-garg) [inline src/app/components/core/system-alert/system-alert.component.ts:110]: To make future proof, in next PR - return "" at the end of this function as default condition and if empty then don't show the corresponding action button

File: src/app/components/core/system-alert/system-alert.component.ts (hunk 5 above)
Comment (MonikaSharma06) [reply src/app/components/core/system-alert/system-alert.component.ts:109]: The message which we get for update is like-

```{ 
        "alert" : {
                     "key"  : "updates",
                   	"message" : "The following updates are available: fledge-south-opcua",
                     "urgency" : "normal",
                     "timestamp" : "2024-01-19 10:45:37.454192"
                }
}

So I checked `updates` text here.

File: src/app/components/core/system-alert/system-alert.component.ts (hunk 4 above)
Comment (MonikaSharma06) [reply src/app/components/core/system-alert/system-alert.component.ts:125]: this is very light in colour, not visible

File: src/app/components/core/system-alert/system-alert.component.html [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+          </div>
+        </div>
+        <div class="columns mb-0 mr-1">
+          <!-- TODO: Show timestamp -->
+          <!-- <div class="column is-1">
+          </div>
+          <div class="column is-pulled-right">
+            <span class="help">1 m ago</span>
+          </div> -->
+          <div class="column pt-0 pb-0" *ngIf="rolesService.hasEditPermissions()">
```
Comment (praveen-garg) [inline src/app/components/core/system-alert/system-alert.component.html:30]: Use ngContainer for such if condition.

**Review by praveen-garg — APPROVED** [review]

**Review by hemantgarg923 — APPROVED** [review]

**Pull request merged**